src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.test.tsx

```diff
@@ -222,3 +237,129 @@ describe('Datasource Management: Create Datasource form', () => {
     expect(component.find(passwordIdentifier).first().props().isInvalid).toBe(false);
   });
 });
+
+describe('Datasource Management: Create Datasource form with different authType configurations', () => {
+  let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
+  const mockSubmitHandler = jest.fn();
+  const mockTestConnectionHandler = jest.fn();
+  const mockCancelHandler = jest.fn();
+
+  /* Scenario 1: Should render the page normally with all authMethod combinations */
+  test('should render normally with all authMethod combinations', () => {
+    const authMethodCombinationsToBeTested = [
+      [noAuthCredentialAuthMethod],
+      [usernamePasswordAuthMethod],
+      [sigV4AuthMethod],
+      [noAuthCredentialAuthMethod, usernamePasswordAuthMethod],
+      [noAuthCredentialAuthMethod, sigV4AuthMethod],
+      [usernamePasswordAuthMethod, sigV4AuthMethod],
+      [noAuthCredentialAuthMethod, usernamePasswordAuthMethod, sigV4AuthMethod],
+    ];
+
+    authMethodCombinationsToBeTested.forEach((authMethodCombination) => {
+      const mockedContext = mockManagementPlugin.createDataSourceManagementContext();
+      mockedContext.authenticationMethodRegistery = new AuthenticationMethodRegistery();
+
+      authMethodCombination.forEach((authMethod) => {
+        mockedContext.authenticationMethodRegistery.registerAuthenticationMethod(authMethod);
+      });
+
+      component = mount(
+        wrapWithIntl(
+          <CreateDataSourceForm
+            handleTestConnection={mockTestConnectionHandler}
+            handleSubmit={mockSubmitHandler}
+            handleCancel={mockCancelHandler}
+            existingDatasourceNamesList={['dup20']}
+          />
+        ),
+        {
+          wrappingComponent: OpenSearchDashboardsContextProvider,
+          wrappingComponentProps: {
+            services: mockedContext,
+          },
+        }
+      );
+
+      const testConnBtn = component.find(testConnectionButtonIdentifier).last();
+      expect(testConnBtn.prop('disabled')).toBe(true);
+    });
```

Comment:
Why are we testing disabled prop of test connection button?

If I understand correctly, we want to test whether form render normally with all authMethod combinations or no, right? Maybe [Snapshot Testing](https://jestjs.io/docs/snapshot-testing) would be best choice here. What do you think?

Reply:
Thanks for the comment. I tried to use snapshot testing and it will snapshot a huge file (over 1000 lines of code).
Since this test case it to verify UX can be rendered successfully, which means once the test able to complete, we should be good.

So I take [this test case](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5916/files#diff-1cadbf80d14e8c1df75fe7485f5e0dbb1157965ef1aa41c7ce22264b91c5a195R93) as a reference, using test connection button shows up to determine page render successfully.

Thanks

Reply:
As this test case is to verify UX we should verify all component especially dropdown. Disable prop of test connection is not proper way to ensure if form is rendered properly or not. Let's not worry size of snapshot file. I'm open to other approach as well, but it should cover rendering pf entire form especially `superselect` component.

Reply:
Test case updated here:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5916/files#diff-1cadbf80d14e8c1df75fe7485f5e0dbb1157965ef1aa41c7ce22264b91c5a195R283

Thanks